source/extensions/filters/common/rbac/engine.h

```diff
@@ -25,7 +25,8 @@ class RoleBasedAccessControlEngine : public Router::RouteSpecificFilterConfig {
    *                   empty map should be used if there are no headers available.
    */
   virtual bool allowed(const Network::Connection& connection,
-                       const Envoy::Http::HeaderMap& headers) const PURE;
+                       const Envoy::Http::HeaderMap& headers,
+                       bool& darklaunch_allowed) const PURE;
```

Comment:
coding wise,  it is more readable to return a struct with two boolean.

Reply:
Or return an enum. 
Alternatively, keep this function the same, but handle the result in filter only.

Reply:
Thanks, talked to @diemtvu offline, will use another POC PR https://github.com/envoyproxy/envoy/pull/3554, which evaluate two rules(enforcement/darklaunch) independently.